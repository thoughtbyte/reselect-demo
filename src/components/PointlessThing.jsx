import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import { toggleBoolean } from "../globalState/pointlessBoolean.js";

class PointlessThing extends Component {
  handleToggle = () => {
    this.props.toggleBoolean(this.props.pointlessBoolean);
  };

  render() {
    console.log("RENDERING PointlessThing");

    return (
      <div>
        <button className="button" onClick={this.handleToggle}>
          Pointless boolean: {this.props.pointlessBoolean.toString()}
        </button>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    pointlessBoolean: state.pointlessBoolean,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ toggleBoolean }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(PointlessThing);
