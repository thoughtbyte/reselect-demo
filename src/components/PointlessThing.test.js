import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import PointlessThing from "./PointlessThing";

const middlewares = []; // in case you need to use one you can just import it and add it here
const mockStore = configureStore(middlewares);
const initialState = {
  pointlessBoolean: false,
};
const store = mockStore(initialState);

describe("PointlessThing", () => {
  test("should match snapshot", () => {
    const pointlessThing = renderer
      .create(
        <Provider store={store}>
          <PointlessThing />
        </Provider>
      )
      .toJSON();

    expect(pointlessThing).toMatchSnapshot();
  });
});
