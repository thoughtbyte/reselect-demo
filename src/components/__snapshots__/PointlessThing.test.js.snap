// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointlessThing should match snapshot 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
  >
    Pointless boolean: 
    false
  </button>
</div>
`;
